src/com/xilinx/rapidwright/util/VivadoTools.java

```diff
@@ -0,0 +1,149 @@
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.xilinx.rapidwright.design.Design;
+
+/**
+ * Utility methods to provide access to vivado and parse logs
+ *
+ */
+public class VivadoTools {
+
+    public static class ReportRouteStatusResult {
+
+        public final int logicalNets;
+        public final int netsNotNeedingRouting;
+        public final int internallyRoutedNets;
+        public final int netsWithNoLoads;
+        public final int implicitlyRoutedPorts;
+        public final int routableNets;
+        public final int unroutedNets;
+        public final int fullyRoutedNets;
+        public final int netsWithRoutingErrors;
+
+        private static int parseLog(List<String> log, String key) {
+            return Integer.parseInt(
+                    VivadoTools.searchVivadoLog(log, key).get(0).replaceAll("[^\\d]", ""));
+        }
+
+        /**
+         * builds a Tcl script to open the Design d in vivado and run the command
+         * `report_route_status`
+         * 
+         * @param d the design to be analyzed
+         */
+        ReportRouteStatusResult(Design d) {
+            final Path workdir = FileSystems.getDefault()
+                    .getPath("vivadoToolsWorkdir" + FileTools.getUniqueProcessAndHostID());
+            File workdirHandle = new File(workdir.toString());
+            workdirHandle.mkdirs();
+            final Path dcp = workdir.resolve("checkpoint.dcp");
+            final Path tclScript = workdir.resolve("tclScript.tcl");
+
+            d.writeCheckpoint(dcp);
+            List<String> lines = new ArrayList<>();
+            lines.add("open_checkpoint " + dcp);
+            lines.add("report_route_status");
+            lines.add("exit");
+            FileTools.writeLinesToTextFile(lines, tclScript.toString());
+            List<String> log = new ArrayList<>();
+            log = VivadoTools.runTcl(workdir.resolve("outputLog.log"), tclScript, true);
+
+            logicalNets = parseLog(log, "# of logical nets");
+            netsNotNeedingRouting = parseLog(log, "# of nets not needing routing");
+            internallyRoutedNets = parseLog(log, "# of internally routed nets");
+            netsWithNoLoads = parseLog(log, "# of nets with no loads");
+            implicitlyRoutedPorts = parseLog(log, "# of implicitly routed ports");
+            routableNets = parseLog(log, "# of routable nets");
+            unroutedNets = parseLog(log, "# of unrouted nets");
+            fullyRoutedNets = parseLog(log, "# of fully routed nets");
+            netsWithRoutingErrors = parseLog(log, "# of nets with routing errors");
+
+            String[] toDel = workdirHandle.list();
+            for (String s : toDel) {
+                File f = new File(workdirHandle.getPath(), s);
+                f.delete();
+            }
+            workdirHandle.delete();
```

Comment:
Would [`FileTools.deleteFolder()`](https://github.com/Xilinx/RapidWright/blob/6b107b77bd85d8edd72da70ecbefb962222ab026/src/com/xilinx/rapidwright/util/FileTools.java#L846) work here?

Reply:
yes it would: https://github.com/Xilinx/RapidWright/pull/684/commits/de5a93a7f84c8597b203adc68b01f0a7dd3d6855